### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by balins**
Hi,
according to our discussions in previous PRs (#260, #279), I've migrated our CI process from Travis CI to GitHub Actions.
The workflow will be triggered on push and pull request, also it is possible to start it manually from "Actions" tab.
Builds and tests are running in the following combination of environments:
```
      matrix:
        os: [ubuntu-latest, windows-latest]
        python: [3.6, 3.7, 3.8, 3.9]
```
Additionally for every combination with `matrix.os == ubuntu-latest`, we build documentation to make sure everything works.

Apart from building and testing, every push to `master` branch of this repo will result in rebuilding the docs and pushing newly generated files to `scikit-fuzzy/scikit-fuzzy.github.io` repo. Live version of the documentation will be available at [https://scikit-fuzzy.github.io](https://scikit-fuzzy.github.io). 
*_(configuration that will be used to build the documentation: (ubuntu-latest, python 3.9))_ 

Minor changes:
- dependencies: update to match the actual environment used in CI
- CONTRIBUTING: replaced instructions for Travis CI with GitHub Actions
- README: add Matplotlib to dependencies, update link to documentation

**Info for maintainers**
@scikit-fuzzy @JDWarner
To make it possible to push the documentation generated by the workflow to another repository, we need to generate a personal access token with full `repo` privileges. The token's value should be stored in a secret for this (_scikit-fuzzy/scikit-fuzzy_) repository under `API_TOKEN_GITHUB` name. If everything works, we can get rid of `gh-pages` branch and deactivate GitHub Pages environment on this repo.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          pip install -e .
+
+      - name: Run tests
+        run: |
+          nosetests -v --exe --with-coverage --cover-package=skfuzzy skfuzzy
+          flake8 --exclude=test_* skfuzzy docs/examples
+      - name: Build docs
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          pip install "decorator>=4.3,<5"
```
Comment (balins) [inline .github/workflows/main.yml:47]: Temporary fix, we need to wait until NetworkX developers deal with decorator 5.x [Link to the issue](https://github.com/networkx/networkx/issues/4718)

Comment (dschult) [reply .github/workflows/main.yml:47]: decorator 5.0.5 fixed this difficulty. So pip and conda and others should pick up the fixed decorator. NetworkX 2.5.1 also pins requirements to decorator <5 if you want that solution instead.

Comment (balins) [reply .github/workflows/main.yml]: Temporary fix removed. Thanks @dschult!

**Comment by balins** [comment]
@scikit-fuzzy @JDWarner Will we merge this?

**Comment by JDWarner** [comment]
@balins thank you for the ping and sorry for the delay, I've added a token with appropriate scope and name through @scikit-fuzzy  - fingers crossed!

**Comment by balins** [comment]
@JDWarner Looking at the logs it seems like there was a GitHub update that forbids using `git://` when connecting to remote repos... The problem is that the currently used version of `gitwash` submodule (`acaa49b5c73a34f2ad52667fd6601f74fecb3b51`) was using it and thus crashed our docs deployment. We should update this submodule to the latest commit from matthew-brett/gitwash – it was updated to use `https://`.

**Comment by balins** [comment]
Adding the following line:
```
    branch = master
```
to `.gitmodules` and adding `--remote` flag to the command initializing `gitwash` submodule in `docs/Makefile` should get the job done.

**Pull request merged**